### ISSUE TITLE
_watch_connectivity_state uses an incorrect error message

### ISSUE THREAD

**Issue by genericmoniker** [grpc/grpc#29848]
### Is your feature request related to a problem? Please describe.

My application makes many calls via gRPC. Sometimes in my logs I see stack traces like this:

```
Traceback (most recent call last):
File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
self.run()
File "/usr/lib64/python3.6/threading.py", line 864, in run
self._target(*self._args, **self._kwargs)
File "/opt/lib64/python3.6/site-packages/grpc/_channel.py", line 914, in _poll_connectivity
time.time() + 0.2)
File "src/python/grpcio/grpc/_cython/_cygrpc/channel.pyx.pxi", line 487, in grpc._cython.cygrpc.Channel.watch_connectivity_state
File "src/python/grpcio/grpc/_cython/_cygrpc/channel.pyx.pxi", line 349, in grpc._cython.cygrpc._watch_connectivity_state
File "src/python/grpcio/grpc/_cython/_cygrpc/channel.pyx.pxi", line 357, in grpc._cython.cygrpc._watch_connectivity_state
ValueError: Cannot invoke RPC: Channel closed!
```
This tells me that I have a problem, but isn't very helpful in narrowing down where it is since the stack trace is entirely grpcio and standard library threading.

### Describe the solution you'd like

It would help narrow down the problem if the exception message could include the gRPC "method" being called. Maybe there is other information available that would also be helpful?

### Additional context

I also run into this in unit testing, where I usually try to mock out any actual RPCs. Occasionally a patch will "miss" and a real RPC will be attempted and I'll see a "ValueError: Cannot invoke RPC: Channel closed!" exception. It isn't always repeatable, though, and I'm using the xdist plugin to pytest which runs the tests in parallel. So I'll see dots for each test as it runs, and a stack trace like the above in the middle of the dots. If I run w/o parallel tests to print each test as it runs, often the problem doesn't happen again (or at least after a few attempts). Again it would help to know which RPC is being attempted.

**Comment by gnossen** [comment]
I agree that this error message isn't great. In fact, even the "Cannot invoke RPC" message seems to be wrong. Unless I'm mistaken, it looks like this code path is only reachable from [`Channel.subscribe`](https://grpc.github.io/grpc/python/_modules/grpc.html#Channel.subscribe). I think the error message should probably be something more like "Unable to monitor channel state: channel closed". That's one thing that should definitely be fixed.

Is it possible that there's a race condition between you calling `channel.subscribe` and `channel.close`?

**Comment by genericmoniker** [comment]
Just knowing that the message is misleading is helpful. 

The race would be that `channel.close` is called followed by `channel.subscribe` on the closed channel? I've got some locking in place intended to prevent that, but that suggests I should dig into that more deeply.

**Comment by gnossen** [comment]
> The race would be that channel.close is called followed by channel.subscribe on the closed channel? I've got some locking in place intended to prevent that, but that suggests I should dig into that more deeply.

Yes. I believe this is what's happening in your code. If not, then there is some other bug in the gRPC stack to track down besides this misleading error message.

For the moment, I'm going to co-opt this bug to track fixing this error message. If you have reason to believe that the error is not caused by the race I described above, please open a separate issue with more information and I will track it down there.
